src/memary/agent/base_agent.py

```diff
@@ -166,8 +185,19 @@ def locate(self, query: str) -> str:
 
     def vision(self, query: str, img_url: str) -> str:
         """Uses computer vision to process the image specified by the image url and answers the question based on the CV results"""
-        img_docs = load_image_urls([img_url])
-        response = self.openai_mm_llm.complete(prompt=query, image_documents=img_docs)
+        query_image_dir_path = Path('query_images')
+        if not query_image_dir_path.exists():
+            Path.mkdir(query_image_dir_path)
+        
+        data = requests.get(img_url).content
+        query_image_path = os.path.join(query_image_dir_path, 'query.jpg')
+        with open(query_image_path, 'wb') as f:
+            f.write(data)
+        image_documents = SimpleDirectoryReader(query_image_dir_path).load_data()
+
+        response = self.mm_model.complete(prompt=query, image_documents=image_documents)
+
+        os.remove(query_image_path) # delete image after use
```

Comment:
Since llava requires a local copy of the image (to my understanding), not sure if creating and removing an image from a `query_images` directory is the best method.

Reply:
Some alternatives to possibly look into: 

- https://huggingface.co/HuggingFaceM4/idefics-9b-instruct (doesn't seem that we have to load prior to analyzing) 
- https://huggingface.co/microsoft/kosmos-2-patch14-224 (still have to load image)